### PR TITLE
PTK support for custom themes

### PR DESCRIPTION
**Pull request by dyuri**
Added support for changing PTK specific styles (non-pygments related, like `autosuggest` or the `completion-menu`) via the new custom styles (#3878).

Using this I took the opportunity to "merge" `PTK_STYLE` (I think it was ptk1 related, so we don't need it anyway) and `PTK2_STYLE` into `DEFAULT_STYLE_DICT` and replace `$PTK_STYLE_OVERRIDES` with a more generic `$XONSH_STYLE_OVERRIDES`.
(As a next step after #2861, I'd be happy if you could take a look @laloch .)

### CODE REVIEW

**Comment by dyuri** [comment]
Sorry, checking the tests.

File: xonsh/environ.py [hunk 1]
```diff
@@ -1708,6 +1699,18 @@ def DEFAULT_VARS():
             "Whether or not to store the ``stdout`` and ``stderr`` streams in the "
             "history files.",
         ),
+        "XONSH_STYLE_OVERRIDES": Var(
```
Comment (scopatz) [inline xonsh/environ.py:1702]: Should this apply to readline & ansi_colors too then?

File: xonsh/environ.py [hunk 2]
```diff
@@ -1203,13 +1201,6 @@ def DEFAULT_VARS():
             "``DEPTH_1_BIT``, ``DEPTH_4_BIT``, ``DEPTH_8_BIT``, ``DEPTH_24_BIT`` "
             "colors. Default is an empty string which means that prompt toolkit decide.",
         ),
-        "PTK_STYLE_OVERRIDES": Var(
```
Comment (scopatz) [inline xonsh/environ.py:1206]: This should probably go through a deprecation cycle of some sort

File: xonsh/style_tools.py [hunk 3]
```diff
@@ -342,24 +337,18 @@ def style_as_faded(template: str) -> str:
...
             Token.Text: "",
             Token.Text.Whitespace: "ansigray",
+            Token.PTK.Aborting: "ansibrightblack",
+            Token.PTK.AutoSuggestion: "ansibrightblack",
+            Token.PTK.CompletionMenu: "bg:ansigray ansiblack",
+            Token.PTK.CompletionMenu.Completion: "",
+            Token.PTK.CompletionMenu.Completion.Current: "bg:ansibrightblack ansiwhite",
+            Token.PTK.Scrollbar.Arrow: "bg:ansiblack ansiwhite bold",
+            Token.PTK.Scrollbar.Background: "bg:ansibrightblack",
+            Token.PTK.Scrollbar.Button: "bg:ansiblack",
```
Comment (scopatz) [inline xonsh/style_tools.py:349]: I think these will then need to be added to all styles.

File: xonsh/environ.py (hunk 1 above)
Comment (dyuri) [reply xonsh/environ.py:1702]: That was the plan, I just forgot it, will try to do it.

File: xonsh/style_tools.py (hunk 3 above)
Comment (dyuri) [reply xonsh/style_tools.py:349]: They should be there due to change in `pyghooks` where I've replaced the old `PTK_STYLE` to `DEFAULT_STYLE_DICT` in the style chainmap.

File: xonsh/environ.py (hunk 1 above)
Comment (dyuri) [reply xonsh/environ.py:1702]: Now it does.

File: xonsh/environ.py (hunk 2 above)
Comment (dyuri) [reply xonsh/environ.py]: Brought it back, and added a deprecation warning if somebody uses it.

**Review by laloch — APPROVED** [review]
LGTM

**Comment by scopatz** [comment]
Thanks @dyuri!

**Comment by scopatz** [comment]
Hmm This seemed to change the $PROMPT color rendering for me on PTK.  FWIW my `$XONSH_COLOR_STYLE = "monokai"`.  Now the prompt now looks like it has more `default` colors.

**Comment by dyuri** [comment]
That's strange, I use `monokai` as well and have not discovered anything (but my PROMPT is rendered by powerline), gonna take a look.

**Comment by laloch** [comment]
This also made `$XONSH_STYLE_OVERRIDES["bottom-toolbar"] = "noreverse"` crash the shell completely
- reverse/noreverse are not supported
- there's no validation

**Comment by laloch** [comment]
I use `monokai` with powerline and my prompt colors are broken too.

**Comment by scopatz** [comment]
Should we revert for the moment?

**Comment by dyuri** [comment]
Yep, please, I'll take a look.

**Comment by dyuri** [comment]
This is strange indeed:
image

**Pull request merged**